plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -46,22 +46,63 @@ export const Reader = ({ entityId, onReady }: Props) => {
   const theme = useTheme<BackstageTheme>();
 
   const techdocsStorageApi = useApi(techdocsStorageApiRef);
-  const [shadowDomRef, shadowRoot] = useShadowDom();
   const navigate = useNavigate();
+  const shadowDomRef = useRef(null);
+  const [loadedPath, setLoadedPath] = useState('');
+  const [atInitialLoad, setAtInitialLoad] = useState(true);
+  const [newerDocsExist, setNewerDocsExist] = useState(false);
 
-  const { value, loading, error } = useAsync(async () => {
+  const {
+    value: isSynced,
+    loading: syncInProgress,
+    error: syncError,
+  } = useAsync(async () => {
+    return techdocsStorageApi.syncEntityDocs({ kind, namespace, name });
+  });
```

Comment:
Reader should only attempt sync if `techdocs.builder` app config is set to `'local'`.

Reply:
Added a comment